ospf6d/ospf6_flood.c

```diff
@@ -95,7 +96,8 @@ void ospf6_lsa_originate(struct ospf6_lsa *lsa)
 
 	/* if the new LSA does not differ from previous,
 	   suppress this update of the LSA */
-	if (old && !OSPF6_LSA_IS_DIFFER(lsa, old)) {
+	if (old && !OSPF6_LSA_IS_DIFFER(lsa, old)
+	    && lsa->header->adv_router != ospf6->router_id) {
```

Comment:
If I'm reading this right, it will allow re-origination for an LSA that hasn't changed so long as it wasn't locally originated. I understand why this is needed for GR, but will this mess up the general case, when we're not actively participating in a GR?

Reply:
Indeed. I modified the code adding a GR-specific special case.